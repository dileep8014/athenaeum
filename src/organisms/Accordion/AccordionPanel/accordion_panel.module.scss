.basic {
  background-color: color('primary-2');
  margin: auto;
  p, ul {
    max-width: 50em;
  }
  p a {
    color: color('primary-1');
    text-decoration: none;

    &:hover {
      text-decoration: underline;
    }
  }

  .toggleSwitch {
    @include typography-b-10();
  }

  @media #{$desktop} {
    max-width: 50em;
  }
}

.panel {
  @include typography-b-8();
  position: relative;
  border-bottom: 1px solid color('neutral-4');
  padding: ru(1) ru(.75);
  &:last-of-type {
    border-bottom: none;
  }

  .title {
    font-size: inherit;
    font-weight: typography-b-weight(true);
    padding-right: ru(1);
    flex-grow: 1;
  }
}

.header {
  display: flex;
  cursor: pointer;
  align-items: baseline;
}

.ordinal {
  position: absolute;
  font-weight: typography-b-weight(true);
  align-self: start;

  // IE 11 displays the ordinal very strangely. Just best to avoid.
  @media all and ( -ms-high-contrast: none ), ( -ms-high-contrast: active ) {
    display: none;
  }
}

.content, .title {
  padding-left: ru(1);
}

.content {
  overflow: hidden;
  height: auto;
  max-height: 0px;
  opacity: 0;
  transition: max-height 350ms ease-out,
              opacity 200ms;

  p:last-of-type {
    margin-bottom: 0;
  }
}

.toggleSwitch {
  width: 4em;
  text-align: right;
}

.content.content-open {
  max-height: 3000px;
  transition: max-height 1s ease-out,
              opacity 200ms ease-in;
  opacity: 1;
}



