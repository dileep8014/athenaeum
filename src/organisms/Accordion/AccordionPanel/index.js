import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import LinkWrapper from 'atoms/LinkWrapper';
import styles from './accordion_panel.module.scss';

export default class AccordionPanel extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.state = {
      open: false
    };
  }

  handleClick() {
    this.setState({
      open: !this.state.open
    });
  }

  createMarkup() {
    return { __html: this.props.body };
  }

  render() {
    const {
      number,
      title,
      type,
      closeText,
      openText
    } = this.props;

    const sectionClasses = cx(styles['panel'], styles[type]);
    const contentClasses = cx({
      [styles['content']]: true,
      [styles['content-open']]: this.state.open
    });

    return (
      <section className={sectionClasses}>
        <header
          className={styles['header']}
          onClick={this.handleClick}
        >
          <span className={styles['ordinal']}>
            {number}.
          </span>
          <h1 className={styles['title']}>
            {title}
          </h1>
          <span className={styles['toggleSwitch']}>
            <LinkWrapper type='secondary'>
              {this.state.open ? closeText : openText}
            </LinkWrapper>
          </span>
        </header>
        <section
          dangerouslySetInnerHTML={this.createMarkup()}
          className={contentClasses}
        />
      </section>
    );
  }
}

AccordionPanel.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
  body: PropTypes.string,
  title: PropTypes.string,
  openText: PropTypes.string,
  closeText: PropTypes.string,
  number: PropTypes.number,
  type: PropTypes.oneOf([ 'basic' ])
};

AccordionPanel.defaultProps = {
  type: 'basic',
};
