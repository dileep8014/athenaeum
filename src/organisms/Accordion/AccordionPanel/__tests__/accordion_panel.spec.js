import React from 'react';

// import sinon from 'sinon';
import renderer from 'react-test-renderer';

import AccordionPanel from '..';

describe('<AccordionPanel />', () => {
  // let defaultProps;
  //
  // beforeEach(() => {
  //   defaultProps = {
  //     // Add default props here
  //   };
  // });

  it('renders correctly', () => {
    const actual = renderer.create(<AccordionPanel />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
