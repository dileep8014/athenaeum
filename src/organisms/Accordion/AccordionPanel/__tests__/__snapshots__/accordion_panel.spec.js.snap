// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionPanel /> renders correctly 1`] = `
<section
  className="panel basic"
>
  <header
    className="header"
    onClick={[Function]}
  >
    <span
      className="ordinal"
    >
      .
    </span>
    <span
      className="toggleSwitch"
    >
      <a
        className="link neutral-1 secondary"
      />
    </span>
  </header>
  <section
    className="content"
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
</section>
`;
